### PR TITLE
Add valid Distinct case for aggregation

### PR DESCRIPTION
**Pull request by mertak-synnada**
## Which issue does this PR close?
-

Closes #.

## Rationale for this change

This change is for satisfying the related conversation: https://github.com/apache/datafusion/pull/11781#discussion_r1702168887

## What changes are included in this PR?

A valid test case for distinct and group-by usage

## Are these changes tested?

Yes

## Are there any user-facing changes?

No

### CODE REVIEW

**Review by mustafasrepo — APPROVED** [review]
LGTM!. Thanks @mertak-synnada for this PR.

File: datafusion/optimizer/src/replace_distinct_aggregate.rs [hunk 1]
```diff
@@ -82,10 +82,11 @@ impl OptimizerRule for ReplaceDistinctWithAggregate {
                 for dep in input.schema().functional_dependencies().iter() {
                     // If distinct is exactly the same with a previous GROUP BY, we can
                     // simply remove it:
-                    if dep.source_indices[..field_count]
-                        .iter()
-                        .enumerate()
-                        .all(|(idx, f_idx)| idx == *f_idx)
+                    if dep.source_indices.len() >= field_count
```
Comment (alamb) [inline datafusion/optimizer/src/replace_distinct_aggregate.rs:85]: is this change required for the test to pass? Or is it a "drive by" cleanup (made while you were making the change already but not required)?

**Review by alamb — APPROVED** [review]
Thank you @mertak-synnada

File: datafusion/optimizer/src/replace_distinct_aggregate.rs (hunk 1 above)
Comment (mertak-synnada) [reply datafusion/optimizer/src/replace_distinct_aggregate.rs:85]: This one was required to make the test pass

Comment (alamb) [reply datafusion/optimizer/src/replace_distinct_aggregate.rs:85]: Nice -- so this also fixes a bug 🚀
